File: app/src/main/java/com/electrosim/LinkPoint.java
package com.electrosim;

import android.graphics.Canvas;
import android.view.MotionEvent;

public class LinkPoint implements LinkPointInterface {

	float x,y,radius;
	ElementInterface container;
	ConnectorInterface connector;
	
	public LinkPoint(float x, float y, float radius, ElementInterface container) {
		super();
		this.x = x;
		this.y = y;
		this.radius = radius;
		this.container=container;
	}


	@Override
	public boolean pointHitTest(MotionEvent event) {
		if(connector == null){
			float myWorldX = container.getX() + x;
			float myWorldY = container.getY() + y;
			float eventX= event.getX();
			float eventY= event.getY();
			float dist = (float) Math.sqrt(Math.pow((eventX - myWorldX), 2)
					+Math.pow(eventY-myWorldY, 2)) ;
			
			
			if(dist<=radius)
				return true;
			
		}
		return false;
		
	}
	
	@Override
	public void setConnector(ConnectorInterface connector) {
		this.connector = connector;
		
	}


	@Override
	public float getY() {
		
		return container.getY() + y;
	}


	@Override
	public float getX() {
		
		return container.getX() + x;
	}


	@Override
	public void draw(Canvas canvas) {
		
		
	}

}

File: app/src/main/java/com/electrosim/LinkPoint.kt
package com.electrosim

import android.graphics.Canvas
import android.view.MotionEvent
import kotlin.math.pow
import kotlin.math.sqrt

open class LinkPoint(
    private val x: Float,
    private val y: Float,
    private val radius: Float,
    private val container: ElementInterface
) : LinkPointInterface {
    
    private var connector: ConnectorInterface? = null
    
    override fun pointHitTest(event: MotionEvent): Boolean {
        if (connector == null) {
            val myWorldX = container.getX() + x
            val myWorldY = container.getY() + y
            val eventX = event.x
            val eventY = event.y
            val dist = sqrt((eventX - myWorldX).pow(2) + (eventY - myWorldY).pow(2))
            
            if (dist <= radius)
                return true
        }
        return false
    }
    
    override fun setConnector(connector: ConnectorInterface?) {
        this.connector = connector
    }

    override fun getY(): Float {
        return container.getY() + y
    }

    override fun getX(): Float {
        return container.getX() + x
    }

    override fun draw(canvas: Canvas) {
        // Empty implementation
    }
} 